src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx

```diff
@@ -65,26 +74,50 @@ type CurriculumVisualiserProps = {
   handleSelectDomain: (year: string, domain: Domain) => void;
   handleSelectSubject: (year: string, subject: Subject) => void;
   handleSelectTier: (year: string, tier: Tier) => void;
+  handleSelectDiscipline: (year: string, discipline: Discipline) => void;
   duplicateUnitSlugs: Set<string>;
   mobileHeaderScrollOffset?: number;
   setUnitData: (unit: Unit) => void;
   setVisibleMobileYearRefID: (refID: string) => void;
 };
 
-export function createProgrammeSlug(
-  unitData?: Unit | null,
-  examboardSlug?: string | null,
+export function isVisibleUnit(
+  yearSelection: YearSelection,
+  duplicateUnitSlugs: Set<string>,
+  year: string,
+  unit: Unit,
 ) {
-  if (unitData?.keystage_slug === "ks4") {
-    return `${unitData.subject_slug}-${unitData.phase_slug}-${
-      unitData.keystage_slug
-    }${unitData.tier_slug ? "-" + unitData.tier_slug : ""}${
-      examboardSlug ? "-" + examboardSlug : ""
-    }`;
+  const s = yearSelection[year];
+  if (!s) {
+    return false;
   }
-  return unitData
-    ? `${unitData.subject_slug}-${unitData.phase_slug}-${unitData.keystage_slug}`
-    : "";
+  const filterBySubject =
+    !s.subject || s.subject.subject_slug === unit.subject_slug;
+  const filterByDiscipline =
+    s.discipline?.id == -1 ||
+    unit.tags?.findIndex((tag) => tag.id === s.discipline?.id) !== -1;
+  const filterByDomain =
+    !s.domain ||
+    s.domain.domain_id === 0 ||
+    s.domain.domain_id === unit.domain_id;
+  const filterByTier =
+    !s.tier || !unit.tier_slug || s.tier?.tier_slug === unit.tier_slug;
+
+  // Look for duplicates that don't have an examboard, tier or subject parent
+  // (i.e. aren't handled by other filters)
+
+  const isDuplicate =
+    unit.examboard === null &&
+    unit.tier === null &&
+    unit.subject_parent === null &&
+    duplicateUnitSlugs.has(unit.slug);
+  return (
+    filterBySubject &&
+    filterByDomain &&
+    filterByTier &&
+    filterByDiscipline &&
+    !isDuplicate
+  );
```

Comment:
It's not made this easy to see, but this is effectively the removal of createProgrammeSlug (It's already in UnitsTab) and the export of isVisibleUnit (so the duplication of that function can be removed from UnitsTab)